adapters/repos/db/migrator.go

```diff
@@ -113,6 +116,105 @@ func (m *Migrator) UpdateClass(ctx context.Context, className string, newClassNa
 	return nil
 }
 
+// UpdateIndex ensures that the local index is up2date with the latest sharding
+// state (shards/tenants) and index properties that may have been added in the
+// case that the local node was down during a class update operation.
+//
+// This method is relevant when the local node is a part of a cluster,
+// particularly with the introduction of the v2 RAFT-based schema
+func (m *Migrator) UpdateIndex(ctx context.Context, incomingClass *models.Class,
+	incomingSS *sharding.State,
+) error {
+	idx := m.db.GetIndex(schema.ClassName(incomingClass.Class))
+
+	{ // add index if missing
+		if idx == nil {
+			if err := m.AddClass(ctx, incomingClass, incomingSS); err != nil {
```

Comment:
Here is the point which I was trying to make:
if idx == nil, it indicates that the class does not exist. In this case, I would expect `UpdateIndex` to be equivalent to `AddClass` (i.e. return after `AddClass()`).  What do you think?

Reply:
Oh ok, I understand. I thought the `incomingSS` contained information for all shards in the node, but it seems that it is just for the `incomingClass`.

In this case, it makes sense to return early here after `AddClass`. I'll make the change!